From 7c26af7b723eae6ac629ea01058553ff025400ab Mon Sep 17 00:00:00 2001
From: jakub <jakub@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 28 Apr 2011 17:01:55 +0000
Subject: [PATCH 192/200] 	PR middle-end/48597
 	* final.c (final_scan_insn): Call dwarf2out_frame_debug even for
 	inline asm.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@173121 138bc75d-0d04-0410-961f-82ee72b054a4

index 1e1424f..7d3ff14 100644
--- a/gcc/final.c
+++ b/gcc/final.c
@@ -2238,6 +2238,11 @@ final_scan_insn (rtx insn, FILE *file, int optimize_p ATTRIBUTE_UNUSED,
 	    location_t loc;
 	    expanded_location expanded;
 
+	    /* Make sure we flush any queued register saves in case this
+	       clobbers affected registers.  */
+	    if (dwarf2out_do_frame ())
+	      dwarf2out_frame_debug (insn, false);
+
 	    /* There's no telling what that did to the condition codes.  */
 	    CC_STATUS_INIT;
 
-- 
1.7.0.4

